platform/android/os_android.cpp

```diff
@@ -387,6 +387,33 @@ void OS_Android::process_touch(int p_what, int p_pointer, const Vector<TouchPos>
 
 			ERR_FAIL_COND(touch.size() != p_points.size());
 
+			if (touch.size() == 1) {
+				Point2 d = (p_points[0].pos - touch.write[0].pos);
+				if (d.x > 5 || d.x < -5 || d.y > 5 || d.y < -5) {
+					Ref<InputEventPanGesture> ev;
+					ev.instance();
+					ev->set_position(p_points[0].pos);
+					ev->set_delta(d);
+					input->parse_input_event(ev);
+					touch.write[0].pos = p_points[0].pos;
+				}
+			} else if (touch.size() == 2) {
+				Point2 v0 = touch[1].pos - touch[0].pos;
+				float l0 = (v0.x * v0.x) + (v0.y * v0.y);
+				Point2 v1 = p_points[1].pos - p_points[0].pos;
+				float l1 = (v1.x * v1.x) + (v1.y * v1.y);
+				float f = (l1 / l0);
```

Comment:
you can't have 2 touch points at the same position,
l0 beeing the length^2 of the vector, it can't be 0.0f.
but sure, it's better to protect against that.